store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/GetTraits.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.federatedstore.operation;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import org.apache.commons.lang3.exception.CloneFailedException;
+
+import uk.gov.gchq.gaffer.operation.io.Output;
+import uk.gov.gchq.gaffer.store.StoreTrait;
+
+import java.util.Map;
+
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreConstants.KEY_OPERATION_OPTIONS_GRAPH_IDS;
+
+/**
+ * An Operation used for getting traits from the FederatedStore.
+ */
+public class GetTraits implements FederatedOperation, Output<Iterable<? extends StoreTrait>> {
```

Comment:
This could be added to core Gaffer. See GetSchema for an example. The FederatedStore could then override the GetTraitsHandler like we do for GetSchemaHandler.